test/Microsoft.Azure.Devices.Edge.Test/IoTEdgeCheck.cs

```diff
@@ -100,11 +100,15 @@ public async Task IoTEdge_check()
                 }
             });
 
-            // TODO: iotedge check currently has 2 legitimate errors in it. When we update the aziot-submodule, the errors
-            // will go away, and we should change this assertion to empty string instead of passing on 2 errors.
-            // This is better than disabling the test, because it still tests iotedge check to some extent, and it will
-            // remind us with a failed run to update this test.
-            Assert.AreEqual(string.Empty, errors_number);
+            // When test runs behind proxy, MQTT and AMQP checks legitimately fail in three different connectivity checks
+            if (Context.Current.EdgeProxy.HasValue)
+            {
+                Assert.AreEqual("6 check(s) raised errors.", errors_number);
```

Comment:
This is kinda hacky. I think it's better to use the `--dont-run` option to `iotedge check` and disable the checks that shouldn't run behind a proxy there.

Reply:
Fixed